Escape the Maze - Text based game

KiranBaliga opened the pull request:
## Pull Request for PyVerse 💡

### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
Escape the matrix - Text based game

- [x] I have provided the issue title.

---

#### Info about the Related Issue
This is a text-based adventure game where players navigate through a randomly generated 5x5 maze to find the exit while collecting treasures and avoiding traps.

- [x] I have described the aim of the project.
Games for fun!
---

#### Name
Kiran Baliga

- [x] I have provided my name.

---

#### GitHub ID

Kiran Baliga

- [x] I have provided my GitHub ID.

---

#### Email ID

baligakiran89@gmail.com

- [x] I have provided my email ID.

---

#### Identify Yourself

SWOC

- [x] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #1146 

- [x] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
N/A

- [x] I have described my changes.

---

#### How Has This Been Tested?
Local Pycompilers! 

- [x] I have described my testing process.

---

#### Checklist
**Please confirm the following:**  
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

Commit:
Create Escapethemaze.py

Commit:
updating Project-Structure.md

Commit:
Create Readme.md

TheChaoticor commented:
![image](https://github.com/user-attachments/assets/be936306-39d5-449c-b64d-578b011af92e)
This is terminating itself ,please check . And also check the checks that it is failing in github

Commit:
Merge branch 'UTSAVS26:main' into main

Commit:
Update Escapethemaze.py

KiranBaliga commented:
@UTSAVS26 Please review the PR now. The code runs perfectly with the online compilers. The game ends with a successful message only after the player reaches (4,4). It also specifies if someone is entering char other than NEWS. @TheChaoticor

TheChaoticor commented:
Please specify which online python compiler.I tried in some of those but is still not working

Commit:
Merge branch 'UTSAVS26:main' into main

Commit:
Update Escapethemaze.py

KiranBaliga commented:
@TheChaoticor I have adjusted the code. It used to work on some interactive compilers and some ended in an error. I have fixed it to run on any compiler. If the compiler does not have interactions very well, i have put automated move feature. I have run on random online compilers like programmiz and codechef and ran locally as well.

Commit:
Merge branch 'UTSAVS26:main' into main

KiranBaliga commented:
@TheChaoticor Please review

KiranBaliga commented:
@UTSAVS26 @TheChaoticor Please review

UTSAVS26 commented:
@TheChaoticor please review this PR and inform me
